ansible/roles/generate-permission-files/tasks/main.yml

```diff
@@ -0,0 +1,172 @@
+    always_run: yes # run during check mode
+    run_once: yes
+    notify: clean mktemp
+    tags:
+      - configuration
+      - plugin
+      - permissions
+      - permissions-sync
+      - permissions-gen
+      - nginx
+
+  - name: Install pip package locally
+    easy_install: name=pip
+    become: yes
+    always_run: yes # run during check mode
+    delegate_to: 127.0.0.1
+
+  - name: install virtualenv using pip locally
+    pip: name=virtualenv
+    become: yes
+    always_run: yes # run during check mode
+    delegate_to: 127.0.0.1
+
+  - name: install python packages using pip locally
+    pip: name={{ item }} virtualenv={{ creds_dir }}/venv
+    always_run: yes # run during check mode
+    with_items:
+       - htpasswd
+       - gspread
+       - oauth2client
+    delegate_to: 127.0.0.1
+
+  - name: Generate permissions file from google spreadsheets when auth_permissions_gspreadsheets is true
+    when: auth_permissions_gspreadsheets is defined and not(auth_permissions_gspreadsheets | trim == '')
+    shell: "{{ creds_dir }}/venv/bin/python {{ role_path }}/files/generate_user_permissions.py -f \"{{ creds_dir }}/oauth\" -p {{ auth_password_column }} -o \"{{ creds_dir }}/permissions.json\"
+        {%- for sheet in auth_permissions_gspreadsheets.split(';') %} -ss \"{{ sheet }}\" {% endfor -%}"
+    delegate_to: 127.0.0.1
+    always_run: yes # run during check mode
+    run_once: yes
+    tags:
+      - configuration
+      - plugin
+      - permissions
+      - permissions-sync
+      - permissions-gen
+      - nginx
+
+  - name: Generate nginx basic auth file if nginx_proxy_enable is set && it requires permissions file locally
+    when: auth_permissions_gspreadsheets is defined and not(auth_permissions_gspreadsheets | trim == '') and 
+          generate_basic_auth_file is defined and generate_basic_auth_file|bool
+    shell: "{{ creds_dir }}/venv/bin/python {{ role_path }}/files/generate_nginx_basic_auth_file.py {{ creds_dir }}/permissions.json {{ creds_dir }}/{{nginx_basic_auth_file_name}}"
+    delegate_to: 127.0.0.1
+    always_run: yes # run during check mode
+    run_once: yes
+    tags:
+      - configuration
+      - permissions
+      - permissions-sync
+      - permissions-gen
+      - nginx
+
+  - name: Copy generated permissions file from google spreadsheets when auth_permissions_gspreadsheets is true
+    when: marathon_auth_enable is defined and marathon_auth_enable|bool and
+          auth_permissions_gspreadsheets is defined and not(auth_permissions_gspreadsheets | trim == '')
+    copy: src="{{ creds_dir }}/permissions.json" dest="{{ marathon_auth_plugin_location }}/user-permissions.json" owner=root group=root mode=0644
+    become: yes
+    tags:
+      - configuration
+      - plugin
+      - permissions
+      - permissions-sync
+      - permissions-gen
+
+  - name: Copy permissions file from local src when marathon_auth_permissions_local_path is defined
```

Comment:
Not all marathon_auth vars. Only the ones that are common for both marathon and nginx have changed.